stack/building-single-package-docs.rst

```diff
@@ -0,0 +1,155 @@
+- :doc:`building-pipelines-lsst-io-with-documenteer-job`
+
+Nevertheless, a single-package build is useful for checking formatting and proof-reading your work.
+This page describes how to run a single-package documentation build, using `pipe_base`_ as the example.
+
+.. _build-package-docs-prereqs:
+
+Prerequisites
+=============
+
+Before starting this tutorial, you’ll need a working ``lsst_distrib`` installation.
+This installation should already be set up with a command like :command:`setup lsst_distrib`.
+
+This installation needs to be a recent daily or weekly build since you’ll be compiling the `pipe_base`_ repository from its ``master`` branch.
+Working from the tip of the ``master`` branch is the norm for LSST software development.
+
+.. _build-package-docs-documenteer:
+
+Install Documenteer, the documentation tooling
+==============================================
+
+Documenteer_ provides tooling to build `pipelines.lsst.io`_.
+Since it’s a `PyPI-distributed Python package <https://pypi.org/project/documenteer/>`__, you need to install it separately from the EUPS Stack.
+The best way to do this is in a Python virtual environment that’s layered on top of the EUPS Stack’s :file:`site-packages`.
+This way it’s easy to delete Documenteer and its dependencies without affecting the Python packages that come with the EUPS Stack.
+
+In a base working directory — not inside a repository directory — create the virtual environment and :command:`pip`-install Documenteer through the :file:`requirements.txt` file for `pipelines_lsst_io`_:
+
+.. code-block:: bash
+
+   python -m venv --system-site-packages --without-pip pyvenv
+   source pyvenv/bin/activate
+   curl https://bootstrap.pypa.io/get-pip.py | python
+   pyvenv/bin/pip install -r requirements.txt
+   curl -O https://raw.githubusercontent.com/lsst/pipelines_lsst_io/tickets/DM-11253/requirements.txt
+   pyvenv/bin/pip install -r requirements.txt
+
+.. todo::
+
+   FIXME update branch after ``tickets/DM-11253`` is done.
+
+.. note::
+
+   By using the :file:`requirements.txt` file in the `pipelines_lsst_io`_ repository, you can ensure you’re using the same version of Documenteer_ and its dependencies as in the CI builds of `pipelines.lsst.io`_.
+
+.. tip::
+
+   When you open a new terminal session, you can reactivate the Python virtual environment in the :file:`pyvenv` directory by running:
+
+   .. code-block:: bash
+
+      source pyvenv/bin/activate
+
+   Do this *after* setting up the EUPS Stack.
+
+.. _build-package-docs-install-package:
+
+Downloading and installing a package
+====================================
+
+You need to clone and build the package locally.
+In this example, you’ll be installing and building `pipe_base`_:
+
+.. code-block:: bash
+
+   git clone https://github.com/lsst/pipe_base
+   cd pipe_base
+   setup -k -r .
+   scons
+
+.. note::
+
+   If you’re actively developing a package, it’s likely that you’ve already installed and built that package.
```

Comment:
"installed" -> "cloned".  Please reserve "install" for the result of `scons install`.